ci/helm-chart/templates/deployment.yaml

```diff
@@ -177,9 +177,12 @@ spec:
         {{- if .existingClaim }}
         persistentVolumeClaim:
           claimName: {{ .existingClaim }}
-        {{- else }}
+        {{- else if .hostPath }}
         hostPath:
           path: {{ .hostPath }}
           type: Directory
+        {{- else }}
+        emptyDir:
+          {{- toYaml .emptyDir | nindent 10 }}
```

Comment:
I am not too familiar with Helm, but if `emptyDir` is not specified then will this error?  Should we do `{{- else if .emptyDir }}` or make `emptyDir` default to `{}`?

Reply:
When emptyDir and all other configuration fields(hostPath etc.) are not set, the generated yaml would be `emptyDir: null`. I just tested that if such yaml is applied it would be transformed to `emptyDir: {}` by k8s.

I am not sure if it is the intended behaviour tho. I'd imagine that one of `existingClaim`|`hostPath`|`emptyDir` has to be set. Now it effectively defaults to emptyDir if not set. Would it be more desirable if an error is thrown when none of the fields above is set?

Reply:
Ahh that makes sense, thank you!

I am not sure either; I could see arguments for either way.  Hmm I vote we keep it as is it currently is.